library/core/game_test/src/main/java/org/quiltmc/qsl/game_test/api/package-info.java

```diff
@@ -0,0 +1,117 @@
+ * 		);
+ * 	}
+ * }
+ * </code></pre>
+ * <p>
+ * <h4>Gradle Setup Example</h4>
+ * Here's an example of Gradle setup to handle tests in a separate sourceSet, with a special task to run the tests:
+ * <pre><code>
+ * sourceSets {
+ * 	testmod
+ * }
+ *
+ * dependencies {
+ * 	// testmod sourceSet should depend on everything in the main source set.
+ * 	testmodImplementation sourceSets.main.output
+ * }
+ *
+ * loom {
+ * 	runs {
+ * 		testmodClient {
+ * 			client()
+ * 			source(sourceSets.testmod)
+ * 		}
+ *
+ * 		testmodServer {
+ * 			server()
+ * 			source(sourceSets.testmod)
+ * 		}
+ *
+ * 		gameTestServer {
+ * 			inherit testmodServer
+ * 			configName = "Game test server"
+ *
+ * 			// Enable the game test runner.
+ * 			property("quilt.game_test", "true")
+ * 			property("quilt.game_test.report_file", "${project.buildDir}/game_test/report.xml")
+ * 			runDir("build/game_test")
+ * 		}
+ * 	}
+ * }
+ *
+ * afterEvaluate {
+ * 	tasks.test.dependsOn tasks.runGameTestServer
+ * }
+ * </code></pre>
+ *
+ * <p>
+ * <h3>Test registration</h3>
+ * While methods annotated with {@link net.minecraft.test.GameTest} in a {@link org.quiltmc.qsl.game_test.api.QuiltGameTest} provided as entrypoint
+ * may be sufficient for some, if you have lots of tests, or need conditional testing, you might want to look into
+ * {@link org.quiltmc.qsl.game_test.api.QuiltGameTest#registerTests(org.quiltmc.qsl.game_test.api.TestRegistrationContext)}, which is a method that
+ * you can override in the entrypoint to provide more tests through the given context.
+ *
+ * <p>
+ * <h3>Structure Handling</h3>
+ * <p>
+ * If {@link net.minecraft.test.GameTest#structureName()} isn't specified, the default structure identifier will be
+ * {@code <mod namespace>:<test class name (lower case)/<test method name (lower case)}.
+ * <p>
+ * If you prefer custom structure paths, but have a common prefix in the class, you can annotate the class with
+ * {@link org.quiltmc.qsl.game_test.api.TestStructureNamePrefix} which accept as value the common prefix,
+ * then in every {@link net.minecraft.test.GameTest} of the class you only need what's following after the prefix for the structure name.
+ *
+ * <p>
+ * <h3>Custom Test Method Invocation</h3>
+ * Test methods are gathered using the {@link net.minecraft.test.GameTest} annotation, but for methods which the class implements
+ * {@link org.quiltmc.qsl.game_test.api.QuiltGameTest} you can override the method
+ * {@link org.quiltmc.qsl.game_test.api.QuiltGameTest#invokeTestMethod(QuiltTestContext, org.quiltmc.qsl.game_test.api.TestMethod)}
+ * which override entirely how the test is invoked, the default invocation is done using
+ * {@link org.quiltmc.qsl.game_test.api.TestMethod#invoke(java.lang.Object, java.lang.Object...)} with the instance of the class (if the method isn't
+ * {@code static}), and as first parameter the test context.
+ */
+
+package org.quiltmc.qsl.game_test.api;
```

Comment:
ew! is snake_case really the convention here?

Reply:
It's not really, and I would push against in most cases

but having a game package just is weird? and I'm worried that a bare test package would make people think they imported something internal?

Thoughts to have

Reply:
The exact thing I tried to avoid because I absolutely despite it as a single word and the actual data directory in data pack for structures use `game_test` :blob_waaaaa:

Reply:
Personally I don't mind using `game_test`. Having `gametest` where we also call it 'Game Test' and 'GameTest' feels wrong.

Reply:
Common convention is to mash everything into one word (mypackage, hackmd, stackoverflow, the list goes on). I don't really understand why we are being different here.

Reply:
I have thought of another package that would fit better being `testing.game`
In retrospect that new package name would fit better and would allow other testing modules in the future and not limiting ourselves to just "game"
Thus increasing package discovery yada yada

tl;dr to me `testing.game` makes more sense now